spec/lucky/text_helpers_spec.cr

```diff
@@ -0,0 +1,343 @@
+      view.pluralize(2, "count", plural: "counters").should eq "2 counters"
+      view.pluralize(nil, "count", plural: "counters").should eq "0 counters"
+      view.pluralize(2, "person").should eq "2 people"
+      view.pluralize(10, "buffalo").should eq "10 buffaloes"
+      view.pluralize(1, "berry").should eq "1 berry"
+      view.pluralize(12, "berry").should eq "12 berries"
+    end
+  end
+
+  describe "cycle" do
+    describe Lucky::TextHelpers::Cycle do
+      it "cycles" do
+        value = Lucky::TextHelpers::Cycle.new("one", 2, "3")
+        value.to_s.should eq "one"
+        value.to_s.should eq "2"
+        value.to_s.should eq "3"
+        value.to_s.should eq "one"
+        value.reset
+        value.to_s.should eq "one"
+        value.to_s.should eq "2"
+        value.to_s.should eq "3"
+      end
+    end
+
+    it "cycles" do
+      view.cycle("one", 2, "3").should eq "one"
+      view.cycle("one", 2, "3").should eq "2"
+      view.cycle("one", 2, "3").should eq "3"
+      view.cycle("one", 2, "3").should eq "one"
+      view.cycle("one", 2, "3").should eq "2"
+      view.cycle("one", 2, "3").should eq "3"
+    end
+
+    it "cycles with array" do
+      array = [1, 2, 3]
+      view.cycle(array).should eq "1"
+      view.cycle(array).should eq "2"
+      view.cycle(array).should eq "3"
+    end
+
+    it "cycle resets with new values" do
+      view.cycle("even", "odd").should eq "even"
+      view.cycle("even", "odd").should eq "odd"
+      view.cycle("even", "odd").should eq "even"
+      view.cycle(1, 2, 3).should eq "1"
+      view.cycle(1, 2, 3).should eq "2"
+      view.cycle(1, 2, 3).should eq "3"
+      view.cycle(1, 2, 3).should eq "1"
+    end
+
+    it "cycles named cycles" do
+      view.cycle(1, 2, 3, name: "numbers").should eq "1"
+      view.cycle("red", "blue", name: "colors").should eq "red"
+      view.cycle(1, 2, 3, name: "numbers").should eq "2"
+      view.cycle("red", "blue", name: "colors").should eq "blue"
+      view.cycle(1, 2, 3, name: "numbers").should eq "3"
+      view.cycle("red", "blue", name: "colors").should eq "red"
+    end
+
+    it "gets current cycle with default name" do
+      view.cycle("even", "odd")
+      view.current_cycle.should eq "even"
+      view.cycle("even", "odd")
+      view.current_cycle.should eq "odd"
+      view.cycle("even", "odd")
+      view.current_cycle.should eq "even"
+    end
+
+    it "gets current cycle with named cycles" do
+      view.cycle("red", "blue", name: "colors")
+      view.current_cycle("colors").should eq "red"
+      view.cycle("red", "blue", name: "colors")
+      view.current_cycle("colors").should eq "blue"
+      view.cycle("red", "blue", name: "colors")
+      view.current_cycle("colors").should eq "red"
+    end
+
+    it "gets current cycle with no exceptions" do
+      view.current_cycle.should be_nil
+      view.current_cycle("colors").should be_nil
+    end
+
+    it "gets current cycle with more than two names" do
```

Comment:
I've been thinking about this a bit, and it sees this test is pretty big. What do you think about splitting it into separate `highlight_spec`, `truncate_spec`, etc?

Reply:
On returning text vs. string. There's probably cases where ```raw``` makes sense as well. The Rails implementation has a escape option on many of the methods to allow for this. Night sure the right answer but we should think about that a bit.